### ISSUE TITLE
Strange code in relevant_intersections

### ISSUE THREAD

**Issue by AlexandrGraschenkov** [MarketaJu/diamond_space#1]
This line looks strange.
https://github.com/MarketaJu/diamond_space/blob/37460abcd9a6aef2d03992decdd00ef50404741a/diamond_space/accumulator.py#L272

It can be simplified to `return points, np.logical_and(np.logical_and(X, Y), regular_points[:,:,0])`.

**Comment by RomanJuranek** [comment]
True, there is no point for the other `logical_and`. Actually I think it could be even more simplified to

```python
return points, np.logical_and.reduce( (X, Y, regular_points[...,0]) )
```

Will push updates if I find some time. Or if you care, you can make a merge request @AlexandrGraschenkov

**Comment by AlexandrGraschenkov** [comment]
It's not crucial for me. I use this code as reference to build `C++` implementation. Not sure it's good choice, maybe better was use original implementation as reference.

How do you build this framework? Do you use original paper as reference? Do you use original code somehow?

Also I found 1 mistake for line weights. `_generate_lines` filter some lines, so indexing in weight array is wrong. To fix you can pass `i` index outside inside tuple `yield`.
https://github.com/MarketaJu/diamond_space/blob/37460abcd9a6aef2d03992decdd00ef50404741a/diamond_space/accumulator.py#L298

**Comment by RomanJuranek** [comment]
@MarketaJu  is the author of the original paper from BMVC. This repo is python re-implementation of the original code from the paper. The old code is [here](https://www.fit.vutbr.cz/research/groups/graph/pclines/pub_page.php?id=2013-BMVC-VanPoints) (Matlab/C++) which you probably already found. Both, the old C++ and this repo are experimental so you can expect some bugs here and there. Also there is a [book](https://dl.acm.org/doi/book/10.5555/2389418) which gives some advises how to implement this method so it may be helpful.
